### PR TITLE
Compute SHA-256 digest for PKCE using the Web Crypto API

### LINKED ISSUE #33250: Compute SHA-256 digest for PKCE using the Web Crypto API
In order to make it easier to serve Keycloak JS without having any external dependencies the `@noble/hashes` library used to compute the SHA-256 digest for PKCE should be replaced with [Web Crypto API](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest).

### PR DESCRIPTION
**Pull request by jonkoops**
Closes #33250

### CODE REVIEW

File: docs/guides/securing-apps/javascript-adapter.adoc
Comment (jonkoops) [inline docs/guides/securing-apps/javascript-adapter.adoc:1]: Note that I also updated all the calls to `init()` here to be `await`ed. This is not related to the code changes made in this PR, but is done for consistency with other methods.

**Comment by jonkoops** [comment]
Putting this on hold as some of the tests in `JavascriptAdapterTest` are failing locally. These are not triggered on CI, hence it looks like everything is passing. I'll report back once they are fixed.

**Comment by mposolda** [comment]
> Putting this on hold as some of the tests in `JavascriptAdapterTest` are failing locally. These are not triggered on CI, hence it looks like everything is passing. I'll report back once they are fixed.

I've just tried to run `JavascriptAdapterTest` locally from my IDE in keycloak main. The 43 tests are passing, but 2 tests are failing (`testSilentCheckSso` and `testSilentCheckSsoNotAuthenticated`).

It looks like a mistake that this test is not tested in the CI? Maybe the option is, that we can first add this test to some CI group (which probably also means fixing the currently failing tests) and then as a follow-up proceed with this PR https://github.com/keycloak/keycloak/pull/33251 ?

**Comment by jonkoops** [comment]
I've added Keycloak JS to the conditional for the CI, so the tests are triggered now when changes are made to it. This seems to trigger the failures. Working on a fix.

**Comment by jonkoops** [comment]
Figured out what the bug was, some parts of the test suite are expecting exceptions when running some code to detect that they are on the login page. Rather strange, but I've restored that code that facilitates this behavior and the tests are now passing again. This can be fixed later when the tests are eventually refactored in the future.

**Review by edewit — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]

**Review by rmartinc — APPROVED** [review]
Changes LGTM!

**Review by mposolda — APPROVED** [review]

**Pull request merged**